.github/workflows/release.yaml

```diff
@@ -3,58 +3,83 @@ on:
   push:
     tags:
       - "v*"
+
 env:
   VERSION: ${{ github.ref_name }}
-  IMAGE: quay.io/jetstack/cert-manager-approver-policy:${{ github.ref_name }}
+
 jobs:
-  docker-image:
+  build_images:
+    runs-on: ubuntu-latest
+
     permissions:
-      contents: write
-      id-token: write
-    runs-on: ubuntu-22.04
+      contents: read # needed for checkout
+      packages: write # needed for push images
+      id-token: write # needed for keyless signing
+
     steps:
       - uses: actions/checkout@v4
-      - uses: docker/setup-qemu-action@v3
-      - uses: docker/setup-buildx-action@v3
+
+      - id: go-version
+        run: |
+          make print-go-version >> "$GITHUB_OUTPUT"
+
       - uses: docker/login-action@v3
         with:
           registry: quay.io
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_PASSWORD }}
-      - uses: docker/build-push-action@v5
+
+      - uses: actions/setup-go@v4
         with:
-          context: .
-          file: ./Dockerfile
-          platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le
-          push: true
-          tags: ${{ env.IMAGE }}
-          build-args: |
-            VERSION=${{ env.VERSION }}
-
-  github-release:
-    runs-on: ubuntu-22.04
-    needs:
-      - docker-image
+          go-version: ${{ steps.go-version.outputs.result }}
```

Comment:
Don't you prefer to point to the `go.mod` file here to automatically build using the latest patch version?
I do.

Reply:
The problem is that the go version is not linked to the source code. Instead, it is linked to the latest version of go at that time.
I think for reproducibility and as a way to properly keep track of versions, we should explicitly version go instead.

Reply:
Turns out that setup-go doesn't necessarily pick up latest version of Go:

* https://github.com/actions/setup-go/issues/434